presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSource.java

```diff
@@ -242,4 +247,16 @@ private static Object deserializePartitionValue(Type type, String valueString, S
         // Iceberg tables don't partition by non-primitive-type columns.
         throw new PrestoException(GENERIC_INTERNAL_ERROR, "Invalid partition type " + type.toString());
     }
+
+    private Block nativeValueToBlock(Type type, Object prefilledValue)
+    {
+        if (prefilledValue != null && type instanceof TimestampType && ((TimestampType) type).getPrecision() == MILLISECONDS) {
+            BlockBuilder blockBuilder = type.createBlockBuilder(null, 1);
+            writeNativeValue(type, blockBuilder, MICROSECONDS.toMillis((long) prefilledValue));
+            return blockBuilder.build();
+        }
+        else {
+            return Utils.nativeValueToBlock(type, prefilledValue);
+        }
```

Comment:
I think we can simplify this a little bit

```suggestion
        if (prefilledValue != null && type instanceof TimestampType && ((TimestampType) type).getPrecision() == MILLISECONDS) {
            return Utils.nativeValueToBlock(type, 
 MICROSECONDS.toMillis((long) prefilledValue));
        }
        return Utils.nativeValueToBlock(type, prefilledValue);
```

Reply:
Fixed, thanks.